src/utils/vSync.h

```diff
@@ -0,0 +1,84 @@
+#ifndef _V_SYNC_H_
+#define _V_SYNC_H_
+#include <qstring.h>
+#include <QObject>
+#include <qprocess.h>
+
+class QMessageBox;
+class QPushButton;
+class VSync : public QObject
+{
+	Q_OBJECT
+private:
+	enum class SyncType
+	{
+		None,
+		Status,
+		Add,
+		Commit,
+		Push,
+		Pull,
+		Authentication
+	};
+
+	enum class SyncTarget
+	{
+		None,
+		Upload,
+		Download,
+	};
+signals:
+	void downloadSuccess();
+	void uploadSuccess();
+public:
+	VSync(QWidget *parent = NULL);
+	~VSync();
+	void setDir(const QString &dir);
+	void upload();
+	void download();
+private slots:
+	void onReadOutput();
+	void onReadError();
+	void onProcessFinish(int exitCode);
+
+private:
+	void status();
+	void add();
+	void commit();
+	void push();
+	void pull();
+	void authentication();
+	void processDownload();
+	void processUpload();
+	void downloadFinish();
+	void uploadFinish();
+
+	void start(const QString &cmd);
+	void showMessageBox(const QString &message, bool showButton);
+	void hideMessageBox();
+	void onMessageButtonClick();
+	QString VSync::getSyncHead(const QString &args) const;
+
+private:
+	QString _dir;
```

Comment:
In VNote, we use `m_xxx` as the name of data members. Anyway, we could leave it unchanged here in this PR.